### ISSUE TITLE
[Feature] Support RETYPE style alerts / admonitions / callouts

### ISSUE THREAD

**Issue by smbambling** [typora/typora-issues#5951]
The GH style alerts introduced in 1.8 are awesome, but a little inflexible as you can't set a title for the callout. 

As someone who uses retype to host a personal wiki it would be awesome to see their style of alerts support which are more robust and flexible allowing a title to be set.

This would allow me to have a better preview when writing docs and also provide an alternative method for using typora to generate html to be used statically if retype is ever abandoned or put behind an extreme paywall 

Reference: https://retype.com/components/alert/

**Comment by smbambling** [comment]
Poke, any acknowledgment of this feature style ?

**Comment by Squirreljetpack** [comment]
As an alternative of this suggestion, maybe allow specify title in brackets part like
![ALERT] instead of ![NOTE].
The styling might default to a special color like gray, or allow to be set in css somehow.
Obsidian callouts does this.

**Comment by smbambling** [comment]
Having the title set in the callout is the biggest win, and I will gladly welcome that in an alternative form.

Selfishly I would like to support RETYPE alerts as they are flexible...and I use the project for my personal sites 🙃

**Comment by HoeYeungHo** [comment]
I agree with this point. Although the same effect can be achieved by modifying the CSS file, it would be even better if it could be directly modified in the markdown file!

**Comment by Haiulus** [comment]
I found that you can change the name of alert in MarkPad. This feature would be awesome in Typora. Icons mabye different but the ability to change the name of alert is great.

<img width="805" height="949" alt="Image" src="https://github.com/user-attachments/assets/858392f0-0398-4d88-b65d-394b0d7659bf" />

**Comment by dpls3757** [comment]
Yeesh – two and a half years later and this is still open. Well, I'll pitch in my two cents anyways, and hope that @abnerlee, @vassudanagunta, @lesander, or @vastbluesky pick it up.

I agree completely with everything that's been expressed here – setting a title (e.g., `[!NOTE] Title`), changing the alert type (e.g., `[!ALERT]`), a tool to allow creating/styling custom (and default) alerts without having to mess directly with the CSS (including custom icons).

There are obviously (and annoyingly) numerous different syntaxes for callouts/admonitions/alerts/whatever you want to call them. While it would be great to have the flexibility of supporting a wide range of them, the first step is supporting one, which was already done with the GitHub alerts, and the next step is expanding that to allow customization beyond the GitHub defaults.

I provide, for reference, [this page](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#admonitions) which has a useful expanded selection of admonitions, as well as a short instruction for creating custom admonitions. I would also just add a vote for this format for callouts – typing "!!! note" requires less distance and less effort than "[!NOTE]".

That format also has very easy and intuitive support for changing or removing the title, nesting admonitions, and collapsible and inline blocks. They also provide a list of alternate icons that can be used.

I hope this feature request will be picked up soon and integrated. It would be very useful indeed!
